### PR TITLE
Mark and UnMark test cases

### PR DESCRIPTION
**Pull request by wnchan**
Finished mark command, unmark command, mark command parser, unmark command parser and added parser util test cases.
Fix [checkstyle main] and [checkstyle test] errors.

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 1]
```diff
@@ -306,6 +307,40 @@ public void parseTutorials_duplicateInput_singleInstanceStored() throws Exceptio
...
+    public void parseTaskIndex_validInput_success() throws Exception {
+        int expectedTaskIndex = 1;
+        int actualTaskIndex = ParserUtil.parseTaskIndex("1");
+        assertEquals(expectedTaskIndex, actualTaskIndex);
+    }
+
+    @Test
+    public void parseTaskIndex_validInputWithWhitespace_success() throws Exception {
+        int expectedTaskIndex = 5;
+        int actualTaskIndex = ParserUtil.parseTaskIndex("  5  ");
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:340]: Follow how the previous withWhitespace test cases were already implemented

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 2]
```diff
@@ -306,6 +307,40 @@ public void parseTutorials_duplicateInput_singleInstanceStored() throws Exceptio
...
+
+    @Test
+    public void parseTaskIndex_validInput_success() throws Exception {
+        int expectedTaskIndex = 1;
+        int actualTaskIndex = ParserUtil.parseTaskIndex("1");
+        assertEquals(expectedTaskIndex, actualTaskIndex);
+    }
+
+    @Test
+    public void parseTaskIndex_validInputWithWhitespace_success() throws Exception {
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:338]:
```suggestion
    public void parseTaskIndex_validInputWithWhitespace_returnsTrimmedTaskIndex() throws Exception {
```

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 3]
```diff
@@ -306,6 +307,40 @@ public void parseTutorials_duplicateInput_singleInstanceStored() throws Exceptio
...
+        assertThrows(ParseException.class, () -> ParserUtil.parseTaskIndex(INVALID_TASK_INDEX_NEGATIVE));
+    }
+
+    @Test
+    public void parseTaskIndex_invalidInputNonNumeric_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseTaskIndex(INVALID_TASK_INDEX_NON_NUMERIC));
+    }
+
+    @Test
+    public void parseTaskIndex_validInput_success() throws Exception {
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:331]: change to method name to withoutWhitespace bc the method below is also a valid input
```suggestion
    public void parseTaskIndex_validValueWithoutWhitespace_returnsTaskIndex() throws Exception {
```

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 4]
```diff
@@ -306,6 +307,40 @@ public void parseTutorials_duplicateInput_singleInstanceStored() throws Exceptio
...
+        int expectedTaskIndex = 1;
+        int actualTaskIndex = ParserUtil.parseTaskIndex("1");
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:333]: Follow how the previous withoutWhitespace test cases were implemented i.e. use constant values

**Review by PearlynnT — COMMENTED** [review]
Can edit ParserUtilTest so that it follows the same format

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+        // No need to use TaskInitializer as tasks are automatically initialised when group is created.
+        model.addGroup(group);
+        model.updateFilteredGroupList(PREDICATE_SHOW_ALL_GROUPS);
+    }
+
+    @Test
+    public void execute_validTaskIndex_success() throws CommandException {
+        int groupId = 1;
+        int taskIndex = 0; // When printing message, CommandResult adds 1 to taskIndex.
+        MarkCommand MarkCommand = new MarkCommand(1, taskIndex,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:42]:
```suggestion
        MarkCommand markCommand = new MarkCommand(1, taskIndex,
```

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+                        + "❌ D 12. CS2103T Finalise TP. 17/11/2023 2359\n"
+                        + "❌ T 13. CS2103T Update DG for each feature. \n",
+                result.getFeedbackToUser());
+    }
+
+    @Test
+    public void execute_taskAlreadyMarked_success() throws CommandException {
+        int groupId = 1;
+        int taskIndex = 0; // Task is already marked
+        MarkCommand MarkCommand = new MarkCommand(groupId, taskIndex,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:68]:
```suggestion
        MarkCommand markCommand = new MarkCommand(groupId, taskIndex,
```

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+                        + "❌ D 12. CS2103T Finalise TP. 17/11/2023 2359\n"
+                        + "❌ T 13. CS2103T Update DG for each feature. \n",
+                result.getFeedbackToUser());
+    }
+
+    @Test
+    public void execute_groupNotFound_throwsCommandException() {
+        int groupId = 2; // Invalid groupId
+        int taskIndex = 0;
+        MarkCommand MarkCommand = new MarkCommand(groupId, taskIndex,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:94]:
```suggestion
        MarkCommand markCommand = new MarkCommand(groupId, taskIndex,
```

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+                new GroupContainsKeywordsPredicate(Arrays.asList(String.valueOf(groupId))));
+
+        assertThrows(CommandException.class, () -> MarkCommand.execute(model));
+    }
+
+    @Test
+    public void execute_invalidTaskIndex_throwsCommandException() {
+        int groupId = 1;
+        int taskIndex = 15; // Invalid task index
+        MarkCommand MarkCommand = new MarkCommand(groupId, taskIndex,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:104]:
```suggestion
        MarkCommand markCommand = new MarkCommand(groupId, taskIndex,
```

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+                new GroupContainsKeywordsPredicate(Arrays.asList(String.valueOf(groupId))));
+
+        assertThrows(CommandException.class, () -> MarkCommand.execute(model));
+    }
+
+    @Test
+    public void execute_nullModel_throwsNullPointerException() {
+        int groupId = 1;
+        int taskIndex = 0;
+        MarkCommand MarkCommand = new MarkCommand(groupId, taskIndex,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:114]:
```suggestion
        MarkCommand markCommand = new MarkCommand(groupId, taskIndex,
```

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+                new GroupContainsKeywordsPredicate(Arrays.asList(String.valueOf(groupId))));
+
+        assertThrows(NullPointerException.class, () -> MarkCommand.execute(null));
+    }
+
+    @Test
+    public void execute_negativeTaskIndex_throwsCommandException() {
+        int groupId = 1;
+        int taskIndex = -1; // Negative task index
+        MarkCommand MarkCommand = new MarkCommand(groupId, taskIndex,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:124]:
```suggestion
        MarkCommand markCommand = new MarkCommand(groupId, taskIndex,
```

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+                new GroupContainsKeywordsPredicate(Arrays.asList(String.valueOf(groupId))));
+
+        assertThrows(CommandException.class, () -> MarkCommand.execute(model));
+    }
+
+    @Test
+    public void isValidTaskIndex_validTaskIndex_returnsTrue() {
+        int taskIndex = 1;
+        TaskList taskList = group.getTasks();
+        MarkCommand MarkCommand = new MarkCommand(1, taskIndex,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:134]:
```suggestion
        MarkCommand markCommand = new MarkCommand(1, taskIndex,
```

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+        boolean isValid = MarkCommand.isValidTaskIndex(taskIndex, taskList);
+
+        assertTrue(isValid);
+    }
+
+    @Test
+    public void isValidTaskIndex_invalidTaskIndex_returnsFalse() {
+        int taskIndex = 15; // Invalid task index
+        TaskList taskList = group.getTasks();
+        MarkCommand MarkCommand = new MarkCommand(1, taskIndex,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:146]:
```suggestion
        MarkCommand markCommand = new MarkCommand(1, taskIndex,
```

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+        boolean isValid = MarkCommand.isValidTaskIndex(taskIndex, taskList);
+
+        assertFalse(isValid);
+    }
+
+    @Test
+    public void equals_sameObject_returnsTrue() {
+        int groupId = 1;
+        int taskIndex = 0;
+        MarkCommand MarkCommand = new MarkCommand(groupId, taskIndex,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:158]:
```suggestion
        MarkCommand markCommand = new MarkCommand(groupId, taskIndex,
```

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java (hunk 1 above)
Comment (wnchan) [reply src/test/java/seedu/address/logic/parser/ParserUtilTest.java:340]: alright!

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+                new GroupContainsKeywordsPredicate(Arrays.asList(String.valueOf(groupId))));
+
+        assertTrue(MarkCommand.equals(MarkCommand));
+    }
+
+    @Test
+    public void equals_differentObjectSameValues_returnsTrue() {
+        int groupId = 1;
+        int taskIndex = 0;
+        MarkCommand MarkCommand1 = new MarkCommand(groupId, taskIndex,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:168]:
```suggestion
        MarkCommand markCommand1 = new MarkCommand(groupId, taskIndex,
```

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 15]
```diff
@@ -0,0 +1,189 @@
...
+                new GroupContainsKeywordsPredicate(Arrays.asList(String.valueOf(groupId))));
+        MarkCommand MarkCommand2 = new MarkCommand(groupId, taskIndex,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:170]:
```suggestion
        MarkCommand markCommand2 = new MarkCommand(groupId, taskIndex,
```

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 16]
```diff
@@ -0,0 +1,189 @@
...
+                new GroupContainsKeywordsPredicate(Arrays.asList(String.valueOf(groupId))));
+
+        assertTrue(MarkCommand1.equals(MarkCommand2));
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:173]:
```suggestion
        assertTrue(markCommand1.equals(markCommand2));
```

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 17]
```diff
@@ -0,0 +1,189 @@
...
+    }
+
+    @Test
+    public void equals_differentObjectDifferentValues_returnsFalse() {
+        int groupId1 = 1;
+        int taskIndex1 = 0;
+        int groupId2 = 2;
+        int taskIndex2 = 1;
+        MarkCommand MarkCommand1 = new MarkCommand(groupId1, taskIndex1,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:182]:
```suggestion
        MarkCommand markCommand1 = new MarkCommand(groupId1, taskIndex1,
```

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 18]
```diff
@@ -0,0 +1,189 @@
...
+                new GroupContainsKeywordsPredicate(Arrays.asList(String.valueOf(groupId1))));
+        MarkCommand MarkCommand2 = new MarkCommand(groupId2, taskIndex2,
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:184]:
```suggestion
        MarkCommand markCommand2 = new MarkCommand(groupId2, taskIndex2,
```

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java (hunk 2 above)
Comment (wnchan) [reply src/test/java/seedu/address/logic/parser/ParserUtilTest.java:338]: thanks!

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 19]
```diff
@@ -0,0 +1,189 @@
...
+                new GroupContainsKeywordsPredicate(Arrays.asList(String.valueOf(groupId2))));
+
+        assertFalse(MarkCommand1.equals(MarkCommand2));
```
Comment (PearlynnT) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:187]:
```suggestion
        assertFalse(markCommand1.equals(markCommand2));
```

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java (hunk 3 above)
Comment (wnchan) [reply src/test/java/seedu/address/logic/parser/ParserUtilTest.java:331]: alirght!

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java (hunk 4 above)
Comment (wnchan) [reply src/test/java/seedu/address/logic/parser/ParserUtilTest.java:333]: alright!

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java (hunk 19 above)
Comment (wnchan) [reply src/test/java/seedu/address/logic/commands/MarkCommandTest.java]: edited, thanks!

**Pull request merged**